<?xml version="1.0"?>
<robot name="leg1_project" xmlns:xacro="http://ros.org/wiki/xacro">
  
  <!-- Fix the base_link -->
  <link name="world"/>

  <joint name="fixed" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
     <origin xyz="0.0 0.0 0.0" rpy="0 0 0" />
  </joint>

<link name="base_link">
    <inertial>
      <origin
        xyz="0.00051804 0.023581 -1.107E-05"
        rpy="0 0 0" />
      <mass
        value="0.043417" />
      <inertia
        ixx="9.9656E-06"
        ixy="-2.555E-07"
        ixz="-5.2331E-09"
        iyy="6.3804E-06"
        iyz="-4.4082E-09"
        izz="1.1002E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://leg1_project_description/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://leg1_project_description/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="link1">
    <inertial>
      <origin
        xyz="-0.00037102 0.035109 0.0004294"
        rpy="0 0 0" />
      <mass
        value="0.083098" />
      <inertia
        ixx="2.7192E-05"
        ixy="1.2278E-06"
        ixz="-1.4412E-08"
        iyy="2.864E-05"
        iyz="-2.5391E-07"
        izz="1.9508E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://leg1_project_description/meshes/link1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://leg1_project_description/meshes/link1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint1"
    type="revolute">
    <origin
      xyz="0 0.03525 0"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="link1" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="1"
      velocity="4.8" />
    <!--dynamics damping ="0.7"/-->
  </joint>
  <link
    name="link2">
    <inertial>
      <origin
        xyz="-0.00096247 0.098146 -0.0027245"
        rpy="0 0 0" />
      <mass
        value="0.25913" />
      <inertia
        ixx="0.00086628"
        ixy="-1.0337E-06"
        ixz="2.4954E-07"
        iyy="5.4494E-05"
        iyz="-1.0491E-05"
        izz="0.00086108" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://leg1_project_description/meshes/link2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://leg1_project_description/meshes/link2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint2"
    type="revolute">
    <origin
      xyz="0 0.079 0.0125"
      rpy="0 0 1.5708" />
    <parent
      link="link1" />
    <child
      link="link2" />
    <axis
      xyz="-1 0 0" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="1"
      velocity="4.8" />
    <!--dynamics damping ="0.7"/-->
  </joint>
  <link
    name="link3">
    <inertial>
      <origin
        xyz="-0.01958 0.088874 -3.9419E-05"
        rpy="0 0 0" />
      <mass
        value="0.071285" />
      <inertia
        ixx="0.00020648"
        ixy="4.9665E-06"
        ixz="-2.2416E-09"
        iyy="6.6111E-06"
        iyz="1.0204E-08"
        izz="0.00020827" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://leg1_project_description/meshes/link3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://leg1_project_description/meshes/link3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint3"
    type="revolute">
    <origin
      xyz="0.0196 0.18555 -0.03025"
      rpy="-1.5708 0 0" />
    <parent
      link="link2" />
    <child
      link="link3" />
    <axis
      xyz="1 0 -0.0012766" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="1"
      velocity="4.8" />
    <!--dynamics damping ="0.7"/-->
  </joint>
</robot>
